flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -254,6 +262,58 @@ public void failJobIfNotEnoughResources() throws Exception {
 				"Could not allocate the required slot within slot request timeout.").isPresent());
 	}
 
+	@Test
+	public void vertexIsResetBeforeRestarted() throws Exception {
+		final JobGraph jobGraph = singleNonParallelJobVertexJobGraph();
+
+		final TestSchedulingStrategy.Factory schedulingStrategyFactory = new TestSchedulingStrategy.Factory();
+		final DefaultScheduler scheduler = createScheduler(jobGraph, schedulingStrategyFactory);
+		final TestSchedulingStrategy schedulingStrategy = schedulingStrategyFactory.getLastCreatedSchedulingStrategy();
+		final SchedulingTopology topology = schedulingStrategy.getSchedulingTopology();
+
+		startScheduling(scheduler);
+
+		final SchedulingExecutionVertex vertex = Iterables.getOnlyElement(topology.getVertices());
```

Comment:
You didn't like this approach?

```
final ArchivedExecutionVertex onlyExecutionVertex = Iterables.getOnlyElement(scheduler.requestJob().getAllExecutionVertices());
final ExecutionAttemptID attemptId = onlyExecutionVertex.getCurrentExecutionAttempt().getAttemptId();
```

`requestJob()` is public API. In contrast, `getExecutionVertex()` is supposed to be only used by subclasses of `SchedulerBase`.

Reply:
`ArchivedExecutionVertex`/`AccessExecutionVertex` does not have the `ExecutionVertexID` field and its getter. Or maybe we should add it?

Reply:
done.